File: 프로그래머스/lv0/120907. OX퀴즈/OX퀴즈.cpp
#include <string>
#include <vector>
#include <sstream>

using namespace std;

vector<string> solution(vector<string> quiz)
{
    vector<string> answer;
    for (int i = 0; i < quiz.size(); i++)
    {
        stringstream ss(quiz[i]);
        int init;
        ss >> init;
        char a, c;
        int b, d;

        while (ss >> a >> b >> c >> d)
        {
            if (a == '+')
            {
                init += b;
                if (init == d) answer.push_back("O");
                else answer.push_back("X");
            }
            else if (a == '-')
            {
                init -= b;
                if (init == d) answer.push_back("O");
                else answer.push_back("X");
            }
        }
    }

    return answer;
}

File: 프로그래머스/0/120907. OX퀴즈/OX퀴즈.java
import java.util.*;

class Solution {
    public String[] solution(String[] quiz) {
        String[] answer = new String[quiz.length];
        
        int idx = 0;
        for (String str : quiz) {
            String[] split = str.split("[ =]+");
            
            int x = Integer.parseInt(split[0]);
            int y = Integer.parseInt(split[2]);
            int z = Integer.parseInt(split[3]);
            
            if (split[1].equals("-")) y *= -1;  
        
            answer[idx++] = x + y == z ? "O" : "X";
        }
        
        return answer;
    }
}